### PR TITLE
[RFC] injector: Use inline Envoy bootstrap config instead of k8s Secret mounted volume

### PR DESCRIPTION
**Pull request by draychev**
TLDR: This PR removes the creation of k8s secret per Envoy sidecar and instead passes the bootstrap config to Envoy via CLI param

---

## Before this PR:
1. a new Pod is about to be created
2. mutating webhook intercepts the creation
3. new bootstrap (xDS) certificate is created for the Envoy
4. certificate (and bootstrap YAML) is saved in a Kubernetes Secret
5. New pod is augmented with Envoy sidecar container
6. Secret is mounted to the sidecar; YAML and cert are exposed via file
7. Envoy (in sidecar container) is pointed to the file `--config-path`

Envoy secrets are left hanging in the K8s cluster forever (as of this writing we have no clean-up mechanism)

## After this PR:

1. new Pod is about to be created
2. mutating webhook intercepts the creation
3. new bootstrap (xDS) certificate is created for the Envoy
4. certificate along with YAML is passed to Envoy via `--config-yaml` CLI param

See:  https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-config-yaml

With this change there is no need to create a Secret per Envoy, which leaves customers K8s cleaner. Also removes code from the repo.

---

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
`no`

### CODE REVIEW

File: pkg/injector/envoy_config.go [hunk 1]
```diff
@@ -1,21 +1,18 @@
 package injector
 
 import (
-	"context"
 	"encoding/base64"
 	"fmt"
 	"strconv"
 
+	ghyaml "github.com/ghodss/yaml"
```
Comment (shashankram) [inline pkg/injector/envoy_config.go:8]: Any reason to alias this import? Would drop the alias if not required.

**Review by shashankram — CHANGES_REQUESTED** [review]
The reason the bootstrap config was a secret instead of command line arguments is so that the bootstrap certificates that are a part of the bootstrap config are not exposed in plaintext as a part of the pod config.

Have you considered the security implications of this change? I'd like to see this commented on before going forward with this change.

**Comment by eduser25** [comment]
though I have concerns on scalability when creating a k8s secret per envoy, I think @shashankram concerns are warranted.

I'll go and suggest an in-between approach, to separate the cert from the config, have the config load through yaml (as this patch introduces), and instead of inlining the cert, load it as [filepath](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-msg-config-core-v3-datasource) by mounting the secret on the envoy.

All this under the assumption we see value in this change (I believe this patch will expose envoy config through inspect, which might make it more accessible)

File: pkg/injector/envoy_config.go (hunk 1 above)
Comment (draychev) [reply pkg/injector/envoy_config.go:8]: Yes - `yaml` is already taken in this context

**Review by ksubrmnn — CHANGES_REQUESTED** [review]
Please update https://github.com/openservicemesh/osm/blob/main/docs/uninstallation_guide.md#remaining-after-osm-uninstallation to remove the second bullet stating that envoy bootstrap config secret would persist post uninstall

**Comment by draychev** [comment]
shelving this conversation
